scipy/sparse/construct.py

```diff
@@ -404,19 +404,26 @@ def _compressed_sparse_stack(blocks, axis):
     """
     other_axis = 1 if axis == 0 else 0
     data = np.concatenate([b.data for b in blocks])
-    indices = np.concatenate([b.indices for b in blocks])
-    indptr = []
+    idx_dtype = get_index_dtype(arrays=[b.indptr for b in blocks],
+                                maxval=data.size)
```

Comment:
The `arrays` should also list the `indices`.
It's possible to imagine very sparse matrices of sizes > 2**32 so also `indices` can overflow.